cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -465,5 +465,36 @@ type ComponentConfig interface {
 	Get() interface{}
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResetConfiguration contains a list of fields that are specifically "kubeadm reset"-only runtime information.
+type ResetConfiguration struct {
+	metav1.TypeMeta
+
+	// CertificatesDir specifies the directory where the certificates are stored. If specified, it will be cleaned during the reset process.
+	CertificatesDir string
+
+	// CleanupTmpDir specifies whether the "/etc/kubernetes/tmp" directory should be cleaned during the reset process.
+	CleanupTmpDir bool
+
+	// CRISocket is used to retrieve container runtime info and used for the removal of the containers.
+	// If CRISocket is not specified by flag or config file, kubeadm will try to detect one valid CRISocket instead.
+	CRISocket string
+
+	// DryRun tells if the dry run mode is enabled, don't apply any change if it is and just output what would be done.
+	DryRun bool
+
+	// Force flag instructs kubeadm to reset the node without prompting for confirmation.
+	Force bool
+
+	// IgnorePreflightErrors provides a slice of pre-flight errors to be ignored during the reset process,
+	// e.g. 'IsPrivilegedUser,Swap', 'all' ignores errors from all checks.
```

Comment:
According to the code, `all` is not supported here.
- `ignorePreflightErrors: Invalid value: "all": 'all' cannot be used in configuration file`

Reply:
i forgot why this was done. history is here:
https://github.com/kubernetes/kubernetes/commit/e706d272629979ffb4a125c5d330496d4b463137#diff-71c615dc50838bd457062893d9f12906d130be2ea2f1ce964317dc43ca82ad2bR477-R481

we could update the comment for IgnorePreflightErrors for init / join / reset to say "all" is not allowed.

we can keep it like that and if someone complains in the future we can have further discussion to remove the "all" vs config limitation.

Reply:
hm...but it's odd given the CLI flag allows it.

how about we just remove the "all" limitation in config everywhere.
WDYT?

Reply:
To be honest, I sometimes have to use the flag to ignore all errors to do some tests on my existing cluster to run `init`. 😄

Reply:
my +1 to remove the limitation and keep CLI / config consistent.
we could just have a warning if user has "all" instead of throwing an error.

Reply:
Good catch! I forgot this limitation in the config, I remember there is a testcase for that, anyway, I will update the comments here and leave this as a follow up.